src/FieldResult.hack

```diff
@@ -38,5 +38,5 @@ final class ValidFieldResult extends FieldResult {
  * The error needs to be propagated to the closest parent nullable field, at which point a ValidFieldResult will be
  * returned, with null value and the same error.
  */
-final class InvalidFieldResult extends FieldResult {
+final class InvalidFieldResult extends FieldResult<nothing> {
```

Comment:
This is the first time I successfully used the bottom type for anything so I am appropriately proud of it.